packages/expo-document-picker/ios/DocumentPickerModule.swift

```diff
@@ -0,0 +1,203 @@
+import ExpoModulesCore
+import UIKit
+import MobileCoreServices
+import Foundation
+
+/**
+ Helper struct storing single picking operation context variables that have their own non-sharable state.
+ */
+struct PickingContext {
+  let promise: Promise
+  let options: DocumentPickerOptions
+  let documentPickerHandler: DocumentPickerHandler
+}
+
+public class DocumentPickerModule: Module, DocumentPickingResultHandler {
+  public func definition() -> ModuleDefinition {
+    Name("ExpoDocumentPicker")
+
+    AsyncFunction("getDocumentAsync") { (options: DocumentPickerOptions, promise: Promise) -> Void in
+      // TODO: do we need this?
+      //  if (promise.resolver != nil) {
+      //    promise.reject(DocumentPickingInProgressException())
+      //  }
+
+      guard let utilities = self.appContext?.utilities else {
+        throw UtilitiesInterfaceNotFoundException()
+      }
+
+      self.launchDocumentPicker(options: options, promise: promise)
+
+      // TODO: ADD try catch above for catching errors about icloud entitlement
+
+    }.runOnQueue(.main)
+  }
+
+  private var currentPickingContext: PickingContext?
+
+  private func launchDocumentPicker(options: DocumentPickerOptions, promise: Promise) {
+    guard let currentViewController = self.appContext?.utilities?.currentViewController() else {
+      return promise.reject(MissingCurrentViewControllerException())
+    }
+
+    let documentPickerDelegate = DocumentPickerHandler(documentPickingResultHandler: self)
+    let picker: UIDocumentPickerViewController
+
+    let mimeTypes = options.type
+
+    if mimeTypes.isEmpty {
+      return promise.reject(IncorrectTypeArgumentException())
+    }
+
+    let shouldCopyToCacheDirectory = options.copyToCacheDirectory
+
+    if #available(iOS 14, *) {
+      var utTypes: [UTType] = []
+
+      for mimeType in mimeTypes {
+        if let convertedUTType = mimeTypeToUTType(mimeType: mimeType) {
+          utTypes.append(convertedUTType)
+        }
+      }
+      picker = UIDocumentPickerViewController.init(forOpeningContentTypes: utTypes)
```

Comment:
Should this not use this initializer?
```swift 
picker = UIDocumentPickerViewController(forOpeningContentTypes: utTypes, asCopy: true)
```

Reply:
Yep, you're right, somehow I missed it. I was really hoping this would solve the issue with copying, but it seems like the issue persists.